ivy/data_classes/array/statistical.py

```diff
@@ -638,6 +638,79 @@ def cumprod(
             out=out,
         )
 
+    def cummin(
+        self: ivy.Array,
+        /,
+        *,
+        axis: int = 0,
+        reverse: bool = False,
+        dtype: Optional[Union[ivy.Dtype, ivy.NativeDtype]] = None,
+        out: Optional[ivy.Array] = None,
+    ) -> ivy.Array:
+        """
+        ivy.Array instance method variant of ivy.cummin. This method simply wraps the
+        function, and so the docstring for ivy.cummin also applies to this method
+        with minimal changes.
+
+        Parameters
+        ----------
+        self
+            input array
+        axis
+            int, axis along which to take the cumulative product. Default is ``0``.
+        exclusive
+            optional bool, whether to exclude the first value of the input array.
+            Default is ``False``.
+        reverse
+            Whether to perform the cummin from last to first element in the selected
+            axis. Default is ``False`` (from first to last element)
+        dtype
+            data type of the returned array. If None, if the default data type
+            corresponding to the data type “kind” (integer or floating-point) of x
+            has a smaller range of values than the data type of x (e.g., x has data
+            type int64 and the default data type is int32, or x has data type uint64
+            and the default data type is int64), the returned array must have the
+            same data type as x. if x has a floating-point data type, the returned array
+            must have the default floating-point data type. if x has a signed integer
+            data type (e.g., int16), the returned array must have the default integer
+            data type. if x has an unsigned integer data type (e.g., uint16), the
+            returned array must have an unsigned integer data type having the same
+            number of bits as the default integer data type (e.g., if the default
+            integer data type is int32, the returned array must have a uint32 data
+            type). If the data type (either specified or resolved) differs from the
+            data type of x, the input array should be cast to the specified data type
+            before computing the product. Default: ``None``.
+        out
+            optional output array, for writing the result to.
+
+        Returns
+        -------
+        ret
+            Input array with cumulatively multiplied elements along the specified axis.
+
+        Examples TODO: CHANGE EXAMPLES AND ABOVE DISCRIPTION
+        --------
+        >>> x = ivy.array([1, 2, 3, 4, 5])
+        >>> y = x.cumprod()
+        >>> print(y)
+        ivy.array([1, 2, 6, 24, 120])
+
+        >>> x = ivy.array([[2, 3], [5, 7], [11, 13]])
+        >>> y = ivy.zeros((3, 2), dtype="int32")
+        >>> x.cumprod(axis=1, exclusive=True, out=y)
```

Comment:
This should be the documentation for `cummin` not `cumprod`

Reply:
this is my first PR. I spent a lot of time on getting the compositional functions right. I completely forgot about the doc strings by the end of it. Let me fix it and request a re-review